### PR TITLE
feat: introduce $.raw API

### PR DESCRIPTION
**Pull request by antongolub**
relates #149 #144 #130

- [x] Tests pass
- [x] Appropriate changes to README are included in PR

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -22,8 +22,9 @@ Bash is great, but when it comes to writing scripts,
 people usually choose a more convenient programming language.
 JavaScript is a perfect choice, but standard Node.js library 
 requires additional hassle before using. The `zx` package provides
-useful wrappers around `child_process`, escapes arguments and
-gives sensible defaults.
+useful wrappers around `child_process`,
+escapes arguments (⚠️ please, see `$.qoute` and `$.raw` notes)
```
Comment (WoLfulus) [inline README.md:26]: typo

Comment (antongolub) [reply README.md]: Fixed. Thanks.

**Comment by antonmedv** [comment]
I’m not sure about this API. It definitely adds more complexity in code and documentation. Extra levels of abstraction is not good. 

Maybe just be more clean in docs and explain how to use nodejs api directly if this is needed.

**Comment by antonmedv** [comment]
I think, let's not add complexity. Let $ always quote. For other use cases, something like this maybe used:
```js
await $`${a.split(' ')}`
```

**Comment by antongolub** [comment]
Complexity always come. Let it be in the utility, not in scripts)
```js
const cmd = condition1 ? 'cmd1' : 'cmd2'
const opt = condition2 ? '--foo=bar --baz="qux quxx"' ? ''

$.raw`${cmd} -a -b -c ${opt}`

$`${`${cmd} -a -b -c ${opt}`.split(' ')}`
```

**Comment by antonmedv** [comment]
$.raw is also unsafe, if user input gets into it.

**Comment by antongolub** [comment]
image

**Comment by harrytran998** [comment]
> $.raw is also unsafe if user input gets into it.

I think safe or unsafe, depends on the context of the problem. When using your command, the generate script looks weird(but worked). So why not expose the API and note the unsafe things to the user know when to use it?

**Pull request closed without merging**